### PR TITLE
Use iframes in legacy widget preview.

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
## Description
This PR updates legacy widget block to use an iframe for the preview.
Using an iframe allows the preview to be more reliable as we can load frontend styles and scripts there.

## How has this been tested?
I used the ultimate social media icons for testing purposes.

## After
Screenshot 2019-03-26 at 18 18 22

## Before
image

### CODE REVIEW

**Comment by youknowriad** [comment]
Thanks for working on this Jorge, I think this should be the way to go to preview widgets. I'm having some errors though

Capture d’écran 2019-03-27 à 9 03 12 AM

Also the "edit" part of the widget is not loading.

(I'm using the site origin widgets)

**Comment by noisysocks** [comment]
What's the status of this one? Looks like it needs a rebase 🙂

**Comment by jorgefilipecosta** [comment]
Hi @noisysocks yes we need to rebase this PR. I don't think we should rebase it before https://github.com/WordPress/gutenberg/pull/15801 is landed, otherwise we will waste effort. PR https://github.com/WordPress/gutenberg/pull/15801 has higher priority. I marked this PR as blocked.

**Comment by draganescu** [comment]
Should we refresh this or close it and make an issue out of it? This is still something we need and a good idea!

**Comment by adamziel** [comment]
I rebased and refreshed this PR. It's still not quite there yet, but it's close.

File: packages/edit-widgets/src/blocks/legacy-widget/edit/widget-preview.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import { addQueryArgs } from '@wordpress/url';
+import { Disabled, FocusableIframe } from '@wordpress/components';
+import { useState } from '@wordpress/element';
+
+function WidgetPreview( { widgetAreaId, attributes, ...props } ) {
+	const DEFAULT_HEIGHT = 300;
+	const HEIGHT_MARGIN = 20;
+	const [ height, setHeight ] = useState( DEFAULT_HEIGHT );
+	const currentUrl = document.location.href;
+	const siteUrl = currentUrl.substr( 0, currentUrl.indexOf( 'wp-admin/' ) );
```
Comment (adamziel) [inline packages/edit-widgets/src/blocks/legacy-widget/edit/widget-preview.js:18]: Anyone aware of any a better way of getting site's URL?

Comment (TimothyBJacobs) [reply packages/edit-widgets/src/blocks/legacy-widget/edit/widget-preview.js:18]: It is available in the REST API index at `wp-json`.

**Review by draganescu — COMMENTED** [review]
This renders good enough! ✔️

**Review by draganescu — APPROVED** [review]

File: packages/edit-widgets/src/blocks/legacy-widget/edit/widget-preview.js (hunk 1 above)
Comment (adamziel) [reply packages/edit-widgets/src/blocks/legacy-widget/edit/widget-preview.js:18]: Actually there is another place in navigation editor that could also use this correction - let's address both in a separate PR

File: packages/edit-widgets/src/blocks/legacy-widget/edit/widget-preview.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	const iframeUrl = addQueryArgs( siteUrl, {
+		widgetPreview: {
+			...attributes,
+			sidebarId: widgetAreaId,
+		},
+	} );
+	return (
+		<Disabled>
+			<FocusableIframe
```
Comment (ellatrix) [inline packages/edit-widgets/src/blocks/legacy-widget/edit/widget-preview.js:27]: Why is `FocusableIframe` used if the `onFocus` prop is not?

**Pull request merged**